build failure on LC toss3 machines

When using the umpire host-config to build chai, i get an error.

The command:
```
cmake -C ../src/tpl/umpire/host-configs/toss_3_x86_64_ib/clang_4_0_0.cmake ..
```
gives:
```
CMake Error at blt/cmake/thirdparty/SetupCUDA.cmake:6 (enable_language):
  The CMAKE_CUDA_COMPILER:

    /usr/tce/packages/cuda/cuda-9.1.85/bin/nvcc

  is not a full path to an existing compiler tool.

  Tell CMake where to find the compiler by setting either the environment
  variable "CUDACXX" or the CMake cache entry CMAKE_CUDA_COMPILER to the full
  path to the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  blt/cmake/thirdparty/SetupThirdParty.cmake:77 (include)
  blt/SetupBLT.cmake:100 (include)
  CMakeLists.txt:102 (include)


-- Configuring incomplete, errors occurred!
See also "/g/g15/settgast/workspace/Codes/geosx/CHAI/build-quartz/CMakeFiles/CMakeOutput.log".
See also "/g/g15/settgast/workspace/Codes/geosx/CHAI/build-quartz/CMakeFiles/CMakeError.log".
```
If disable cuda, I get the following error:
```
CMake Error: Error required internal CMake variable not set, cmake may not be built correctly.
Missing variable is:
CMAKE_CUDA_COMPILE_WHOLE_COMPILATION
CMake Error: Error required internal CMake variable not set, cmake may not be built correctly.
Missing variable is:
CMAKE_CUDA_DEVICE_LINK_EXECUTABLE
-- Generating done
-- Build files have been written to: /g/g15/settgast/workspace/Codes/geosx/CHAI/build-quartz
```
I get this last error on many platforms using various hostconfigs. The August20 snapshot doesn't have these problems.

@rhornung67 You are correct. The first part of the issue is to point out that the toss3 hostconfigs for chai/umpire include CUDA by default.

The second part of the issue is the real problem. I get the
```
CMake Error: Error required internal CMake variable not set, cmake may not be built correctly.
Missing variable is:
CMAKE_CUDA_COMPILE_WHOLE_COMPILATION
CMake Error: Error required internal CMake variable not set, cmake may not be built correctly.
Missing variable is:
CMAKE_CUDA_DEVICE_LINK_EXECUTABLE
```
for LC, and when I try to do the GEOSX docker builds for Travis...and on my mac. Any suggestions about that issue?

--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -41,10 +41,19 @@
 # POSSIBILITY OF SUCH DAMAGE.
 #######################################################################
 
+set (chai_umpire_example_depends
+  chai)
+
+if (ENABLE_CUDA)
+  set (chai_umpire_example_depends
+    ${chai_umpire_example_depends{
+    cuda)
+endif()
+
 blt_add_executable(
   NAME chai-umpire-example.exe
   SOURCES chai-umpire-allocators.cpp
-  DEPENDS_ON cuda chai)
+  DEPENDS_ON ${chai_umpire_example_depends})
 
 if (ENABLE_CUDA)
   blt_add_executable(

--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -46,7 +46,7 @@ set (chai_umpire_example_depends
 
 if (ENABLE_CUDA)
   set (chai_umpire_example_depends
-    ${chai_umpire_example_depends{
+    ${chai_umpire_example_depends}
     cuda)
 endif()
 

